lib/tasks/metrics.rake

```diff
@@ -0,0 +1,30 @@
+namespace :metrics do
+  desc 'Exports metrics in all the popular formats to s3'
+  task export: :environment do
+    abort 'Please set the env-var S3_METRICS_BUCKET_NAME' if ENV['S3_METRICS_BUCKET_NAME'].blank?
+
+    # get list of all available metrics
+    METRIC_CLASSES = [Metrics::Moves].map { |namespace|
+      namespace.constants.map { |c| namespace.const_get(c) if namespace.const_get(c).is_a? Class }.compact
+    }.flatten
+
+    s3 = Aws::S3::Resource.new
+    bucket = s3.bucket(ENV['S3_METRICS_BUCKET_NAME'])
+
+    METRIC_CLASSES.each do |metric_class|
+      metric_class.new.tap do |metric|
+        puts metric.label
+        metric_class::FORMATS.each do |format_key, format_file|
+          bucket.object("#{metric.file}/#{format_file}").tap do |obj|
```

Comment:
something to keep in mind but not necessarily for this iteration: is there a need to keep multiple versions of the data or maybe as a fail safe at least one older version, as we can append a timestamp to the filename meaning we don't overwrite older files, and keep a maximum of 2 (or more depending on the need)

Reply:
spoke in person but for posterity: S3 supports file versioning where we have the ability to turn that on if need be, so we don't need to hand roll our own implementation